### PR TITLE
Remote Logs not showing up for API requests

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
Using the api requests, records from remoteLogs were not showing up in the UI.
Handled this edge case.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/results/segresults/segresults.go [hunk 1]
```diff
@@ -498,6 +498,27 @@ func (sr *SearchResults) GetRemoteInfo(remoteID string, inrrcs []*utils.RecordRe
...
+	rawLogs := sr.remoteInfo.remoteLogs
+	count := 0
+	for i := 0; i < len(inrrcs); i++ {
+		if inrrcs[i].SegKeyInfo.IsRemote {
+			finalLogs = append(finalLogs, rawLogs[inrrcs[i].SegKeyInfo.RecordId])
+			count++
+		}
+	}
+
+	return finalLogs
```
Comment (Macbeth98) [inline pkg/segment/results/segresults/segresults.go:519]: You are also supposed to return `remoteCols`.  And assign them in `searchHandler.go` at Line: 404.

File: pkg/segment/segexecution.go [hunk 2]
```diff
@@ -439,6 +439,11 @@ func ExecuteQuery(root *structs.ASTNode, aggs *structs.QueryAggregators, qid uin
 		log.Errorf("qid=%d, ExecuteQuery: failed to get number of RRCs for qid! Error: %v", qid, err)
 	}
 
+	res.RemoteLogs, err = query.GetAllRemoteLogs(res.AllRecords, qid)
```
Comment (Macbeth98) [inline pkg/segment/segexecution.go:442]: This is fine. 
But, I am not sure whether we should fetch the `RawLogs` here itself, it does not exactly match with our current pattern, where we fetch the raw logs in `searchhandler` or `wsSearchHandler`.

**Review by Macbeth98 — COMMENTED** [review]
I see that you have mentioned this as a rough Draft, so its fine for testing. But when are actually merging this, it might better to do this:
 
We could refactor: `GetRemotInfo` function so that we could use for both cases.

As of now, there is only difference I see which is the [`prefix` check happening here](https://github.com/siglens/siglens/pull/1633/files#diff-153fd9c9d5bb1a446f897390942735ad9010e6656c9fcd80494b41cd5ccc5252R477-R486).

So may be we could pass a flag that will say `fetchAllLogs`. And if that is set true we won't check this condition and add all raw logs. And you could send empty string for `remoteId`.

File: pkg/segment/segexecution.go (hunk 2 above)
Comment (vasukalariya) [reply pkg/segment/segexecution.go:442]: This was done to reduce changes in the executeQuery since it is being used at quite a few places and we would need to get remote logs before query gets deleted.

File: pkg/segment/results/segresults/segresults.go (hunk 1 above)
Comment (vasukalariya) [reply pkg/segment/results/segresults/segresults.go]: marked as TODO

File: pkg/segment/query/querystatus.go [hunk 3]
```diff
@@ -558,7 +558,20 @@ func GetRemoteRawLogInfo(remoteID string, inrrcs []*utils.RecordResultContainer,
...
+
+func GetAllRemoteLogs(inrrcs []*utils.RecordResultContainer, qid uint64) ([]map[string]interface{}, []string, error) {
+	arqMapLock.RLock()
+	rQuery, ok := allRunningQueries[qid]
+	if !ok {
+		log.Errorf("GetAllRemoteLogs: qid %+v does not exist!", qid)
+		arqMapLock.RUnlock()
+		return nil, nil, fmt.Errorf("qid does not exist")
+	}
+	defer arqMapLock.RUnlock()
```
Comment (AndrewHess) [inline pkg/segment/query/querystatus.go:572]: nit: your unlocking looks correct, but a bit messy. If you only need the lock for `rQuery, ok := allRunningQueries[qid]`, then you can unlock it (not defer unlocking it) right after you read from that map. Or if you need it for longer, you can defer unlocking it right after you lock it

**Review by AndrewHess — COMMENTED** [review]
Looking good; one nit

File: pkg/segment/query/querystatus.go (hunk 3 above)
Comment (vasukalariya) [reply pkg/segment/query/querystatus.go:572]: oh sorry. Missed it

Comment (vasukalariya) [reply pkg/segment/query/querystatus.go]: addressed in the latest commit. I have corrected this at a few more places. Later, will go over through this file and see if any more could be cleaned up.

**Review by AndrewHess — APPROVED** [review]
Looks good
